frappe/api/v2.py

```diff
@@ -115,6 +118,76 @@ def delete_doc(doctype: str, name: str):
 	return "ok"
 
 
+def add_attachment(doctype: str, name: str):
+	frappe.form_dict.update({"doctype": doctype, "docname": name})
+	return upload_file()
+
+
+def add_tag(doctype: str, name: str):
+	from frappe.desk.doctype.tag.tag import add_tag
+
+	return add_tag(frappe.form_dict.tag, doctype, name)
+
+
+def remove_tag(doctype: str, name: str):
+	from frappe.desk.doctype.tag.tag import remove_tag
+
+	return remove_tag(frappe.form_dict.tag, doctype, name)
+
+
+def get_assignments(doctype: str, name: str):
+	from frappe.desk.form.assign_to import get
+
+	return get({"doctype": doctype, "name": name})
```

Comment:
You'll likely have to perform some perm check. API endpoints are not "@whitelist" so they need to do this manually.